test/amoc_coordinator_SUITE.erl

```diff
@@ -3,40 +3,41 @@
 -include_lib("eunit/include/eunit.hrl").
 -include_lib("common_test/include/ct.hrl").
 
--compile(export_all).
+-compile([export_all, nowarn_export_all]).
 
 -define(MOCK_MOD, mock_mod).
 -define(TELEMETRY_HANDLER, telemetry_handler).
--define(TELEMETRY_HANDLER_CONFIG, #{dummy_config => true}).
 
 
 all() ->
-    [execute_plan_without_timeout,
+    [
+     plan_normalises_successfully,
+     failing_action_does_not_kill_the_worker,
+     execute_plan_without_timeout,
      reset_plan_without_timeout,
-     execute_plan_with_timeout].
+     execute_plan_with_timeout
+    ].
 
 init_per_suite(Config) ->
     meck:new(?MOCK_MOD, [non_strict, no_link]),
     meck:expect(?MOCK_MOD, f_1, ['_', '_'], ok),
     meck:expect(?MOCK_MOD, f_2, ['_', '_', '_'], ok),
     meck:expect(?MOCK_MOD, f_3, ['_', '_', '_', '_'], ok),
-    meck:new(?TELEMETRY_HANDLER, [non_strict, no_link]),
-    meck:expect(?TELEMETRY_HANDLER, handler, ['_', '_', '_', '_'], ok),
-    application:start(telemetry),
-    TelemetryEvents = [[amoc, coordinator, Event] ||
-                           Event <- [start, stop, timeout, reset, add]],
-    TelemetryHandler = fun ?TELEMETRY_HANDLER:handler/4,
-    telemetry:attach_many(?TELEMETRY_HANDLER, TelemetryEvents,
-                           TelemetryHandler, ?TELEMETRY_HANDLER_CONFIG),
-    Config.
+    meck:expect(?MOCK_MOD, f_throw, fun(_, _) -> throw(error) end),
+    TelemetryEvents = [[amoc, coordinator, Event] || Event <- [start, stop, timeout, reset, add]],
+    telemetry_helpers:start(TelemetryEvents),
+    Pid = spawn(fun() -> amoc_coordinator_sup:start_link(), receive terminate -> ok end end),
+    [{sup, Pid} | Config].
 
-end_per_suite(_Config) ->
+end_per_suite(Config) ->
     application:stop(telemetry),
+    Sup = ?config(sup, Config),
+    Sup ! terminate,
```

Comment:
that would be nice to call telemetry_helpers:stop() here, even if we do meck:unload() later